ts/screens/wallet/WalletHomeScreen.tsx

```diff
@@ -142,8 +144,29 @@ class WalletHomeScreen extends React.PureComponent<Props> {
     return this.props.navigation.getParam("newMethodAdded");
   }
 
+  get navigationKeyFrom() {
+    return this.props.navigation.getParam("keyFrom");
+  }
+
   private navListener?: NavigationEventSubscription;
 
+  private handleBackPress = () => {
```

Comment:
You could consider to use **foldL** (I didn't test it)

```typescript
 private handleBackPress = () => {
    fromPredicate((cond: boolean) => cond)(this.newMethodAdded).foldL(
      () => {
        this.props.navigateBack();
      },
      _ => {
        fromNullable(this.props.navigation.getParam("keyFrom")).foldL(
          () => {
            this.props.navigation.setParams({ newMethodAdded: false });
            this.props.navigateToWalletList();
          },
          k => {
            this.props.navigateBack(k);
          }
        );
      }
    );
    return true;
  };
```

Reply:
yep! I like it without IF statements, i just tested and seems to work fine